daemon/list.go

```diff
@@ -137,6 +138,14 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			Names: names[container.ID],
 		}
 		newC.Image = container.Config.Image
+		if container.Config.Image != "" {
+			img, err := daemon.repositories.LookupImage(container.Config.Image)
+			if err == nil {
+				if stringid.TruncateID(img.ID) == stringid.TruncateID(container.Config.Image) {
```

Comment:
@duglin  This is just to check if the `container.Config.Image` is a ID or a name, if it is a ID, then just return the full 64-length ID.
I agree this is not a good idea.